informers.go

```diff
@@ -23,14 +34,89 @@ func getVirtCli() (kubecli.KubevirtClient, error) {
 	return virtCli, err
 }
 
-func getVmiInformer() (cache.SharedIndexInformer, error) {
-	virtCli, err := getVirtCli()
-	if err != nil {
-		return nil, err
-	}
-
+func getVmiInformer(virtCli kubecli.KubevirtClient) (cache.SharedIndexInformer, error) {
 	listWatcher := cache.NewListWatchFromClient(virtCli.RestClient(), "virtualmachineinstances", k8sv1.NamespaceAll, fields.Everything())
 	vmiInformer := cache.NewSharedIndexInformer(listWatcher, &k6tv1.VirtualMachineInstance{}, 1*time.Hour, cache.Indexers{})
+	
+	vmiInformer.AddEventHandler(cache.ResourceEventHandlerFuncs {
+		DeleteFunc: removeWarningLabel,
+	})
 
 	return vmiInformer, nil
 }
+
+func updateMachineTypes(virtCli kubecli.KubevirtClient) error {
+	vmList, err := virtCli.VirtualMachine(k8sv1.NamespaceAll).List(&k8sv1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, vm := range vmList.Items {
+		machineType := vm.Spec.Template.Spec.Domain.Machine.Type
```

Comment:
I've tried to fetch this but neither [VirtualMachineStatus](https://pkg.go.dev/kubevirt.io/api/core/v1#VirtualMachineStatus) nor [VirtualMachineInstanceStatus](https://pkg.go.dev/kubevirt.io/api/core/v1#VirtualMachineInstanceStatus) have a field for machine types.

Reply:
You can find it [here](https://github.com/kubevirt/kubevirt/blob/v1.0.0-rc.0/staging/src/kubevirt.io/api/core/v1/schema.go#L387). As mentioned in [this comment](https://github.com/prnaraya/mass-machine-type-transition/pull/2#discussion_r1212794849), this is how it looks like:
```yaml
spec:
  domain:
    machine:
      type: q35
status:
  machine:
    type: pc-q35-rhel8.2.0
```

Reply:
> so VMs with a machine type of just "q35" will be ignored

In my opinion this is a bug.
What if the VMIs I've created were created at a time where `8.2.0` was the default and latest version? Then you're going to ignore these, although they need to be updated.

Am I missing anything?

Reply:
Ah I see what you mean. I still am not sure how to access the VMI status.machine.type because I just don't see it in any of the documentation anywhere, but it might be good then in the case of the machine type being `q35` to update the machine type anyway to be safe.